### ISSUE TITLE
Still has the issue about creating Window in Sierra 10.12.4

### ISSUE THREAD

**Issue by kesperado** [glfw/glfw#1002]
Just run the simple code provided on the official site

I have tried download the latest version from GitHub, and compiled locally, but seems still has the same problem. I run the code on Xcode:

```
#include <GLFW/glfw3.h>

int main(void)
{
    GLFWwindow* window;
    
    /* Initialize the library */
    if (!glfwInit())
        return -1;
    
    glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 3);
    glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 2);
    glfwWindowHint(GLFW_OPENGL_FORWARD_COMPAT, GL_TRUE);
    glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_CORE_PROFILE);
    
    /* Create a windowed mode window and its OpenGL context */
    window = glfwCreateWindow(640, 480, "Hello World", NULL, NULL);
    if (!window)
    {
        glfwTerminate();
        return -1;
    }
    
    /* Make the window's context current */
    glfwMakeContextCurrent(window);
    
    /* Loop until the user closes the window */
    while (!glfwWindowShouldClose(window))
    {
        /* Render here */
        glClear(GL_COLOR_BUFFER_BIT);
        
        /* Swap front and back buffers */
        glfwSwapBuffers(window);
        
        /* Poll for and process events */
        glfwPollEvents();
    }
    
    glfwTerminate();
    return 0;
}
```

The results shows:
```
2017-05-03 19:52:39.874168-0700 learnOpenGL[5286:355180] [General] ERROR: Setting <GLFWContentView: 0x10060b640> as the first responder for window <GLFWWindow: 0x100418110>, but it is in a different window ((null))! This would eventually crash when the view is freed. The first responder will be set to nil.
(
	0   AppKit                              0x00007fffa7d65a3f -[NSWindow _validateFirstResponder:] + 557
	1   AppKit                              0x00007fffa74e0b1c -[NSWindow _setFirstResponder:] + 31
	2   AppKit                              0x00007fffa7588d9e -[NSWindow _realMakeFirstResponder:] + 406
	3   AppKit                              0x00007fffa7588bb4 -[NSWindow makeFirstResponder:] + 123
	4   libglfw.3.dylib                     0x00000001000bd86e createNativeWindow + 1230
	5   libglfw.3.dylib                     0x00000001000bd14f _glfwPlatformCreateWindow + 63
	6   libglfw.3.dylib                     0x00000001000b59c7 glfwCreateWindow + 887
	7   learnOpenGL                         0x0000000100000e87 main + 135
	8   libdyld.dylib                       0x00007fffbf134235 start + 1
)
```

**Comment by kesperado** [comment]
I used glfwGetVersionString()
and got:
`3.2.1 Cocoa NSGL chdir menubar retina dynamic`

I did downloaded the latest code from GitHub, and use cmake, make, and sudo make install to compile it. But why it is still 3.2.1? instead of 3.3 which seems possible to solve the problem

**Comment by kesperado** [comment]
Would anyone please tell me how to compile the latest version from GitHub? I think I did missed some command

**Comment by elmindreda** [comment]
If you didn't install GLFW 3.2.1 the same way as 3.3 then it may be in a another directory with a higher priority in the linker's list of directories. If you installed it with a package manager then uninstall it using that tool. If not, just search your system for anything `"libglfw*"`.

**Comment by kesperado** [comment]
@elmindreda I searched "libglfw*", found that in:
/usr/local/lib, there is **libglfw.3.2.dylib**
but **libglfw.3.3.dylib** does not exist.

I installed glfw3.2 using cmake, and uninstall it, then install the GitHub version, there is still no **libglfw.3.3.dylib** generated in /usr/local/lib,  Although there also generated .h files in /usr/local/include

The command I used for installing is:
```
cd <glfw-root-dir>
cmake .
mkdir glfw-build
cd glfw-build
cmake <glfw-root-dir>

```
And I plused:
`sudo make install`

I used the same command for both glfw 3.2.1 and glfw-master
But the first one generated dylib in /usr/local/lib, which can be added into Xcode,
While the second master version cannot generate the dylib. Which cause the link error.

So, would anybody tell me what's the right way to install it, so that I can use Xcode to run it
I would really appreciate it!
